src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,257 @@
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        User user = new User();
+        user.setLogin(GOOD_LOGIN_FIRST);
+        user.setPassword(GOOD_PASSWORD_FIRST);
+        user.setAge(AGE_OF_TWENTY);
+        Storage.people.add(user);
+        for (int i = 0; i < 5; i++) {
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_FIRST);
+        user2.setLogin(GOOD_LOGIN_SECOND);
+        user3.setLogin(GOOD_LOGIN_THIRD);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(null);
+        user2.setAge(null);
+        user3.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(null);
+        user2.setLogin(null);
+        user3.setLogin(null);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_FIRST);
+        user2.setLogin(GOOD_LOGIN_SECOND);
+        user3.setLogin(GOOD_LOGIN_THIRD);
+        user1.setPassword(null);
+        user2.setPassword(null);
+        user3.setPassword(null);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
```

Comment:
The `register_nullPassword_notOk` test can be simplified in the same way as the previous tests; use one User object and set the password to null to avoid redundancy.